### PR TITLE
[SPARK-29565][ML][PYTHON] OneHotEncoder should support single-column input/output

### PR DESCRIPTION
**Pull request by huaxingao**
### What changes were proposed in this pull request?
add single-column input/ouput support in OneHotEncoder 

### Why are the changes needed?
Currently, OneHotEncoder only has multi columns support.  It makes sense to support single column as well.

### Does this PR introduce any user-facing change?
Yes
```OneHotEncoder.setInputCol```
```OneHotEncoder.setOutputCol```

### How was this patch tested?
Unit test

### CODE REVIEW

**Review by srowen — COMMENTED** [review]
It seems OK for consistency. I guess however you can already specify just one input column? and is it more usual to one-hot encode probably all categoricals columns at once? I don't mind either way: making it consistent, or arguing this is a bit different and leaving it.

**Comment by huaxingao** [comment]
@srowen Thanks for your comment. I prefer to add the single column support for the consistency and completeness of the APIs. Also, in the use cases of one-hot encoding a single column, it's simpler and more convenient to use an API of one input column.

**Review by srowen — APPROVED** [review]

File: mllib/src/main/scala/org/apache/spark/ml/feature/OneHotEncoder.scala [hunk 1]
```diff
@@ -123,6 +132,14 @@ class OneHotEncoder @Since("3.0.0") (@Since("3.0.0") override val uid: String)
   @Since("3.0.0")
   def this() = this(Identifiable.randomUID("oneHotEncoder"))
 
+  /** @group setParam */
+  @Since("3.0.0")
+  def setInputCol(values: String): this.type = set(inputCol, values)
+
+  /** @group setParam */
+  @Since("3.0.0")
+  def setOutputCol(values: String): this.type = set(outputCol, values)
```
Comment (viirya) [inline mllib/src/main/scala/org/apache/spark/ml/feature/OneHotEncoder.scala]: nit: values -> value.

**Review by viirya — APPROVED** [review]

**Comment by zhengruifeng** [comment]
@huaxingao You need to rebase this PR.

File: mllib/src/main/scala/org/apache/spark/ml/feature/OneHotEncoder.scala (hunk 1 above)
Comment (huaxingao) [reply mllib/src/main/scala/org/apache/spark/ml/feature/OneHotEncoder.scala]: updated. Thanks!

**Comment by viirya** [comment]
Thanks! Merging to master.

**Comment by huaxingao** [comment]
Thanks! @viirya @srowen @zhengruifeng

**Pull request closed without merging**